testhelpers/core.go

```diff
@@ -184,3 +187,158 @@ func RequireCreateStorages(ctx context.Context, t *testing.T) (state.Tree, vm.St
 
 	return st, vms
 }
+
+type testStorage struct {
+	state interface{}
+}
+
+var _ exec.Storage = testStorage{}
+
+func (ts testStorage) Put(v interface{}) (cid.Cid, error) {
+	return cid.Cid{}, nil
+
+}
+func (ts testStorage) Get(cid.Cid) ([]byte, error) {
+	node, err := cbor.WrapObject(ts.state, types.DefaultHashFunction, -1)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return node.RawData(), nil
+}
+
+func (ts testStorage) Commit(cid.Cid, cid.Cid) error {
+	return nil
+}
+
+func (ts testStorage) Head() cid.Cid {
+	return cid.Cid{}
+}
+
+// FakeVMContext creates the scaffold for faking out the vm context for direct calls to actors
+type FakeVMContext struct {
+	TestMessage        *types.Message
+	TestStorage        exec.Storage
+	TestBalance        types.AttoFIL
+	TestBlockHeight    *types.BlockHeight
+	TestVerifier       exec.Verifier
+	TestRandomness     []byte
+	TestIsAccountActor bool
+	Sender             func(to address.Address, method string, value types.AttoFIL, params []interface{}) ([][]byte, uint8, error)
+	Addresser          func() (address.Address, error)
+	Charger            func(cost types.GasUnits) error
+	Sampler            func(sampleHeight *types.BlockHeight) ([]byte, error)
+	ActorCreator       func(addr address.Address, code cid.Cid, initalizationParams interface{}) error
+}
+
+var _ exec.VMContext = &FakeVMContext{}
+
+// NewFakeVMContext fakes the state machine infrastructure so actor methods can be called directly
+func NewFakeVMContext(message *types.Message, state interface{}) *FakeVMContext {
+	randomness := MakeRandomBytes(32)
```

Comment:
Please use a deterministic value, seeded from a parameter if necessary. Nondeterministic behaviour in tests makes debugging hard. Or maybe just leave this nil until a specific test calls for it.